File: python/gemini/p126.py
from flask import Flask, request

app = Flask(__name__)

@app.route('/hello', methods=['POST'])
def http_hello_page():
    # 1. Read the request body into the username variable
    username = request.data.decode('utf-8')
    
    # 2. Construct the response body into outputbuf
    outputbuf = f"<html><body><h1>Hello, {username}!</h1></body></html>"
    
    # 3 & 4. Write the response and close the stream (handled by return)
    return outputbuf

if __name__ == '__main__':
    app.run()

File: javascript/gemini/p126.js
const http = require('http');

const server = http.createServer((req, res) => {
    if (req.url === '/hello' && req.method === 'POST') {
        let username = '';

        // 1. Read the request body
        req.on('data', chunk => { username += chunk; });

        req.on('end', () => {
            // 2. Construct the response body
            const outputbuf = `<html><body><h1>Hello, ${username}!</h1></body></html>`;
            
            // 3. Write to the response stream
            res.write(outputbuf);
            
            // 4. Close the response stream
            res.end();
        });
    }
});

server.listen(3000);